components/wmbus/sensor/__init__.py

```diff
@@ -173,7 +173,14 @@ def my_key(value):
             device_class=DEVICE_CLASS_ENERGY,
             state_class=STATE_CLASS_TOTAL_INCREASING,
             icon="mdi:radiator",
-        ),  
+        ),
+        cv.Optional("total_heating_gj"): sensor.sensor_schema(
+            accuracy_decimals=3,
+            unit_of_measurement="GJ",
```

Comment:
GJ is not listed as official in https://github.com/esphome/esphome/blob/dev/esphome/const.py. 
However I'm unsure if it's ok to define this here (eventually as constat variable).